src/LabelComponent.jsx

```diff
@@ -0,0 +1,158 @@
+import { useState, useEffect } from "react";
+import "./LabelComponent.css";
+
+const LabelComponent = () => {
+  const [inputValue, setInputValue] = useState("");
+  const [labelsData, setLabelsData] = useState([]);
+  const [filteredSuggestions, setFilteredSuggestions] = useState([]);
+  const [selectedLabels, setSelectedLabels] = useState([null]);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch(
+          `http://localhost:3000/api/labels?query=${inputValue}`
+        );
+
+        if (!response.ok) {
+          throw new Error("API request failed");
+        }
+
+        const data = await response.json();
+        setLabelsData(data);
+        handleInputChange(inputValue);
+      } catch (error) {
+        setError(error.message);
+      }
+    };
+
+    fetchData();
+  }, [inputValue]);
+
+  const handleInputChange = (input) => {
+    if (!labelsData || !labelsData.member) {
+      return;
+    }
+    const filtered = input.trim()
+      ? labelsData.member
+          .filter(
+            (label) =>
+              label.name.toLowerCase().includes(input.toLowerCase()) &&
+              label.visibility === "visible"
+          )
+          .map((label) => label.name)
+      : [];
+
+    setFilteredSuggestions(filtered);
+  };
```

Comment:
This seems redundant.
You can do the filtering immediately after the fetch, no need for a seperate function...